src/metabase/driver.clj

```diff
@@ -1215,6 +1248,38 @@
   dispatch-on-initialized-driver
   :hierarchy #'hierarchy)
 
+(defmulti upload-promotion-allowlist
```

Comment:
Getting deep in the nits here, but a suggested rename:

```suggestion
(defmulti allowed-promotions
```

1. I can't see any reason we'd want to artificially limit which promotions are allowed on a driver-by-driver bases. The relaxation rules in CSV restrict us to everything that makes sense, and then each driver should just be best effort.
2. It's not a list.
3. Natural english word order. Highlights the noun.
4. Avoids jargon.

Reply:
821398f227d9